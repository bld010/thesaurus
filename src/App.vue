<template>
  <div id="app">
    <Header />
    <Search v-on:onSearchInput="updateSearchTerm" v-on:receiveResultsFromSearch="saveResults"/>
    <WordList :searchTerm="searchTerm" />
    <h2>App.searchTerm: {{ searchTerm }}</h2>
  </div>
</template>

<script>

import Search from './components/Search.vue';
import Header from './components/Header.vue';
import WordList from './components/WordList.vue';

export default {
  name: 'app',
  components: {
    Header,
    Search,
    WordList
  },
  data: () => {
    return {
      searchTerm: '',
      results: []
    }
  },
  methods: {
    updateSearchTerm (value) {
      this.searchTerm = value
    },
    saveResults(value) {
      this.results = value
    }
  }
}

</script>

<style>

#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  /* margin-top: 60px; */
  margin: 0;
  padding: 0;
}

</style>
