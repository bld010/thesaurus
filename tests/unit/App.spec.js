import App from '../../src/App.vue';
import { shallowMount } from '@vue/test-utils';
import '@vue/cli-plugin-unit-jest';

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallowMount(App);

    expect(wrapper).toMatchSnapshot();
  })
})