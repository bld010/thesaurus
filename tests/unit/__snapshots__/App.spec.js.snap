// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div id="app">
  <header-stub></header-stub>
  <search-stub searchtermfromclick=""></search-stub>
  <wordlist-stub synonyms="" searchterm=""></wordlist-stub>
</div>
`;
